Taken from Oracle solaris-userland b83a793,
components/x11/xserver/xorg/patches/15-6721634.patch (xorg-server 21.1.24).

Solaris-specific changes for:
Bug 15491721 - SUNBT6721634 The Nvidia logo screen should *not* be displayed

Not suitable for upstream

--- xorg-server-21.1.20/hw/xfree86/common/xf86AutoConfig.c
+++ xorg-server-21.1.20/hw/xfree86/common/xf86AutoConfig.c
@@ -61,6 +61,18 @@ "Section \"Device\"\n" \
 	"\tIdentifier\t" BUILTIN_DEVICE_NAME "\n" \
 	"\tDriver\t\"%s\"\n"
 
+#ifdef sun
+/*
+ * Allow setting an arbitrary number of options.
+ * Each option should be indented with a tab and newline terminated.
+ */
+#define BUILTIN_DEVICE_SECTION_PRE_OPT \
+       "Section \"Device\"\n" \
+       "\tIdentifier\t" BUILTIN_DEVICE_NAME "\n" \
+       "\tDriver\t\"%s\"\n" \
+       "%s"
+#endif
+
 #define BUILTIN_DEVICE_SECTION_POST \
 	"EndSection\n\n"
 
@@ -68,6 +80,13 @@ #define BUILTIN_DEVICE_SECTION \
 	BUILTIN_DEVICE_SECTION_PRE \
 	BUILTIN_DEVICE_SECTION_POST
 
+#ifdef sun
+/* Device section with options */
+#define BUILTIN_DEVICE_SECTION_OPT \
+       BUILTIN_DEVICE_SECTION_PRE_OPT \
+       BUILTIN_DEVICE_SECTION_POST
+#endif
+
 #define BUILTIN_SCREEN_NAME \
 	"\"Builtin Default %s Screen %d\""
 
@@ -184,6 +203,14 @@ }
     listPossibleVideoDrivers(&md);
 
     for (i = 0; i < md.nmatches; i++) {
+#ifdef sun
+        const char *nvidia_string = "nvidia";
+        const char *nvidia_opt_no_logo = "\tOption \"NoLogo\" \"True\"\n";
+        if (!strcmp(md.matches[i], nvidia_string))
+            snprintf(buf, sizeof(buf), BUILTIN_DEVICE_SECTION_OPT,
+                      md.matches[i], 0, md.matches[i], nvidia_opt_no_logo);
+        else
+#endif
         snprintf(buf, sizeof(buf), BUILTIN_DEVICE_SECTION,
                 md.matches[i], 0, md.matches[i]);
         AppendToConfig(buf);
